plugin/restore/mysql/mysql.go

```diff
@@ -107,38 +114,45 @@ func (r *Restore) SwapPITRDatabase(ctx context.Context, database string, timesta
 	renameStmt := fmt.Sprintf("RENAME TABLE %s;", strings.Join(tableRenames, ", "))
 
 	if _, err := txn.ExecContext(ctx, renameStmt); err != nil {
-		return err
+		return "", "", err
 	}
 
 	if err := txn.Commit(); err != nil {
-		return err
+		return "", "", err
 	}
 
-	return nil
+	return pitrDatabaseName, pitrOldDatabase, nil
 }
 
-// DeleteOldDatabase deletes the old database after the PITR swap task.
-func (r *Restore) DeleteOldDatabase(ctx context.Context, database string, suffixTs int64) error {
+// DeletePITRDatabases deletes the temporary pitr databases after the PITR swap task.
+func (r *Restore) DeletePITRDatabases(ctx context.Context, database string, suffixTs int64) error {
 	db, err := r.driver.GetDbConnection(ctx, "")
 	if err != nil {
 		return err
 	}
 
-	pitrOldDatabase := getPITROldDatabaseName(database, suffixTs)
-	if _, err := db.ExecContext(ctx, fmt.Sprintf("DROP DATABASE `%s`;", pitrOldDatabase)); err != nil {
+	pitrOldDatabaseName := getPITROldDatabaseName(database, suffixTs)
+	if _, err := db.ExecContext(ctx, fmt.Sprintf("DROP DATABASE `%s`;", pitrOldDatabaseName)); err != nil {
+		return err
+	}
+
+	pitrDatabaseName := getPITRDatabaseName(database, suffixTs)
+	if _, err := db.ExecContext(ctx, fmt.Sprintf("DROP DATABASE `%s`;", pitrDatabaseName)); err != nil {
```

Comment:
I assume the database should contain no table at this point?

If that's the case, then we should check that's the case before proceeding.